src/NHibernate/Cache/ReadWriteCache.cs

```diff
@@ -24,11 +41,20 @@ namespace NHibernate.Cache
 	/// </remarks>
 	public partial class ReadWriteCache : IBatchableCacheConcurrencyStrategy
```

Comment:
The summary comment states:

> Repeatable read isolation is compromised in the case of concurrent writes.

Which means implementing correctly minimal put may cause more violations of repeatable read semantic.
But since it was not guaranteeing it anyway, and since not putting when minimal put is false causes bugs, bette implement it anyway I think.